src/main/java/org/datadog/jmxfetch/reporter/StatsdReporter.java

```diff
@@ -38,14 +38,15 @@ private void init() {
         /* Create the StatsDClient with "entity-id" set to "none" to avoid
            having dogstatsd server adding origin tags, when the connection is
            done with UDS. */
-        log.info("Initializing Statsd reporter with parameters host={} port={} telemetry={} "
-                        + "queueSize={} entityId={}",
+        log.info("Initializing blocking Statsd reporter with parameters host={} port={} "
+                        + "telemetry={} queueSize={} entityId={}",
                 this.statsdHost, this.statsdPort, this.telemetry, this.queueSize, entityId);
         NonBlockingStatsDClientBuilder builder = new NonBlockingStatsDClientBuilder()
                 .hostname(this.statsdHost)
                 .port(this.statsdPort)
                 .enableTelemetry(this.telemetry)
                 .queueSize(this.queueSize)
+                .blocking(true)
```

Comment:
It's a simply change and most likely the default we want, however, we have many Agents out there defaulting to the contrary for now. Because of that, should we consider having a configuration flag if for some reasons we would want to get back to non-blocking on some deployments?